### ISSUE TITLE
Model Release Discussions 2

### ISSUE THREAD

**Issue by lllyasviel** [lllyasviel/style2paints#207]
Because the previous thread is out-of-date and many new previews are released, new thread starts here.

**Comment by ShinkoNet** [comment]
I will leave out most of my questions until I see the workflow-oriented assistant and supporting color scribbles sections being completed in the document.
Thanks to the Dorothy model, I'm fine with the program being called style2paints v5. This is very exciting.

Regarding functionality:
Will Dorothy model support color hints and exporting the color layer only (without lineart)?
How well can it turn a messy sketch/lineart into a completed lineart using the Alice model and prompting?
Can it do manga-style hatching and shading?
If you plan to allow people to draw the lineart directly in the program, the program should use something that isn't gradio. It should use a framework that better supports drawing tools for artists, like your [original SEPA program](https://mobile.twitter.com/lvminzhang/status/1392143022221975554).

Regarding ethics:
The final model release should have all of anything v3's weights replaced with standard stable diffusion. Using a variant of novelai is the most controversial part of the project due to its origin, so it would be best to release the model without its reliance.
The model should also use the same ethical licenses as Stable Diffusion does, e.g the RAIL license.
Considering how beneficial this can be for artists, where it performs best under human-drawn lineart (and thus, human ideas), I see no problems releasing this to the public with the above suggestions.

**Comment by My12123** [comment]
Will the program be available in one language? I would like Style2paints5 to be available in many languages (International languages These include:
Englishman
Spanish
Arabian
Russian
French
Chinese). I am interested in Russian. If yes, what?

**Comment by SwayStar123** [comment]
Will there be model distillation done to reduce inference times?

Is there any work being done on lineart -> lineart? Lineart would be easier to edit and would prove to be a good middle step before going straight to colorization (for the alice method, lineart -> lineart would provide greater control whilst still requiring much less detailed original input)

**Comment by Ihateyoudattebayo** [comment]
Will we ever find out if SEPA is simple or studio level.(https://mobile.twitter.com/lvminzhang/status/1392143022221975554).

**Comment by drbobo0** [comment]
amazing work keep it up

**Comment by lllyasviel** [comment]
Update:
Just like style2paints V4, style2paints V5 will not officially support manga panels. However, just like style2paints V4, users may try unofficial ways to handle manga panels. Handling both illustrations and manga panels is too time-consuming for a small research group.

**Comment by lllyasviel** [comment]
About Node Graph System:

We are *very* afraid of adding a node graph system. 

To the best of our knowledges, adding a node graph system seems always bring bad luck to many different projects. Many projects die out quickly after adding a node system. I can tell a lot examples and I trust you can also name a few. This seems because that node graph has essential violations to fundamental HCI principles and conceptual model rules.

Many users of Blender have some levels of concerns to simplify the node system. They have also made some changes to directly put some buttons inside a right click menu to use, rather than asking users to build a node graph.

Nuke is another exception but they are not really only using node graphs. Their graph system is more like UML diagrams. Nevertheless, even Nuke is less intuitive than Adobe After Effects, which simply put buttons on the interface. See also the market share between Nuke and AE.

In conclusion, we will not add node system before we absolutely need it. We believe this decision helps us escape from the “curse of node graph”, which has already killed many many projects.

**Comment by Ihateyoudattebayo** [comment]
> About Node Graph System:
> 
> We are _very_ afraid of adding a node graph system.
> 
> To the best of our knowledges, adding a node graph system seems always bring bad luck to many different projects. Many projects die out quickly after adding a node system. I can tell a lot examples and I trust you can also name a few. This seems because that node graph has essential violations to fundamental HCI principles and conceptual model rules.
> 
> Many users of Blender have some levels of concerns to simplify the node system. They have also made some changes to directly put some buttons inside a right click menu to use, rather than asking users to build a node graph.
> 
> Nuke is another exception but they are not really only using node graphs. Their graph system is more like UML diagrams. Nevertheless, even Nuke is less intuitive than Adobe After Effects, which simply put buttons on the interface. See also the market share between Nuke and AE.
> 
> In conclusion, we will not add node system before we absolutely need it. We believe this decision helps us escape from the “curse of node graph”, which has already killed many many projects.

Thanks for clarifying.

**Comment by lllyasviel** [comment]
Upd:

V5 Dorothy has potential to become a “clean” method that is free from copyright concerns and AI-generated labels, since it only blend color and minor texture to human drawn contents – all shapes are still drawn by human.

Nevertheless, we still need to take a look at different legal standards of different countries. We will let you know once we have any good news.

**Comment by ycyngot** [comment]
Will it be possible to run V5 on cloud GPU somehow, or local installation is the only way? This software looks very promising, just was wondering if it will be available for users like me with AMD or not very powerful laptops/PC's overall :)

**Comment by lllyasviel** [comment]
Our first round of technical preview is now finished! We should not have any new tweets for the next few weeks. We look forward to seeing you next time.

**Comment by lllyasviel** [comment]
Update:

The release of Style2Paints V5 will be delayed again. We need to make more thoughtful considerations about technology and the potential effects. 

We are committed to presenting technology in a safe and beneficial way for all artists. We will let you know when we are ready. Thank you for your support and understanding!

**Comment by JoshKasap** [comment]
If possible it would be great if v5 supported any models the user wanted and had access too. Making Alice and Dorothy the default and officially supported models is fine, but allowing users to mess around and test other models would be great.

**Comment by Ihateyoudattebayo** [comment]
> If possible it would be great if v5 supported any models the user wanted and had access too. Making Alice and Dorothy the default and officially supported models is fine, but allowing users to mess around and test other models would be great.

That way the community could train the models to be used for manga.

**Comment by pokeball99** [comment]
> We are committed to presenting technology in a safe and beneficial way for all artists. We will let you know when we are ready. Thank you for your support and understanding!

please, whatever you do, do not cause the quality of your excellent work to dip trying to please luddites that want your destruction.

**Comment by Samuelr58** [comment]
PLEASE create a discord server as it will speed up response times for certain issues people are having. Thank You!

**Comment by Lime-Cakes** [comment]
If there is concern about model release, could paper/technical report/code be released first? Information like training method, hyper parameter, and model architecture is of interest to many people. 

Releasing information for academic purpose should be safe, as other companies, research team, and individuals have done.

**Comment by lllyasviel** [comment]
UPD:
We are actively working on releasing technical materials. Please stay tuned.

**Comment by Lime-Cakes** [comment]
> UPD: We are actively working on releasing technical materials. Please stay tuned.

Looking forward to it!

If possible, please include details about the hand crafted weights/layers. Specifically, the ones mentioned below.
> some layer weights are manually written by Lvmin, typically some special connection layers.

The paragraph on [README.md](https://github.com/lllyasviel/style2paints/blob/master/V5_preview/README.md) wasn't clear about how the weight are chosen. Reading it seem to suggest some weights were not trained, but decided through other means, with human supervision and without actually training them through the usual gradient descend? I'm unsure if I'm reading it right. It'd be really, really nice to have clarification about how the mentioned manually written process works.

**Comment by wpfnlp** [comment]
> UPD: We are actively working on releasing technical materials. Please stay tuned.

Will Style2Paints V5 be released this week?

**Comment by sodayo9** [comment]
It seems that they already released everything in 

https://github.com/lllyasviel/ControlNet

**Comment by wpfnlp** [comment]
您的邮件已收到！

**Comment by BigMeatyClaws** [comment]
There is absolutely no reason not to release the lineart model, unless you intend to monetize it.

**Comment by Ihateyoudattebayo** [comment]
> There is absolutely no reason not to release the lineart model, unless you intend to monetize it.

Did you forget about other users training the model.

**Comment by xITmasterx** [comment]
> UPD: We are actively working on releasing technical materials. Please stay tuned.

I have to ask something real quick regarding the release of V5. When is it going to be released? And if not, are there any other issues that needed to be discussed regarding this topic?

**Comment by Avidya1997** [comment]
Wow this update is amazing! As an artist who has tested out the earlier versions of Style2paints this seems like a massive leap in quality/usefulness! 

Dorothy would be so helpful in coloring in the webtoons I’m working on. Alice could be very useful if it can be trained with an input character concept art sheet, so it can call a custom character consistently, like you did with “Rem” example. This would make it so a webtoon could be created even from a rough storyboard vs finalizing line art. 
I guess that can be true for Dorothy too, where it knows which character is which based on the line art so it can use the same color palette consistently without human guidance. 

For the legal issues, I think your caution is warranted as we’ve recently seen stable diffusion and github copilot come under suits from artists/coders/getty images, etc. 

I wonder if a solution can be created where the liability is on the user of the tool, similar to Elevenlabs Voice AI https://beta.elevenlabs.io/terms. You can upload your own voice and have it create a clone of it. Similarly could Style2Paints allow artists to submit their own works to be trained on so it can be used by said artist. Maybe giving a license to Style2Paints of their copyrighted work so they can use the tool and Style2Paint can train on it. 

Maybe in the future there can be a database that artists opt into so their art can be used for training or even get paid a small fee every time it is used.

**Comment by williamkmlau** [comment]
> Regarding ethics: The final model release should have all of anything v3's weights replaced with standard stable diffusion. Using a variant of novelai is the most controversial part of the project due to its origin, so it would be best to release the model without its reliance. As you mentioned in the previous thread, I am glad to hear that this change will not affect the model greatly. The model should also use the same ethical licenses as Stable Diffusion does, e.g the RAIL license. Considering how beneficial this can be for artists, where it performs best under human-drawn lineart (and thus, human ideas), I see no problems releasing this to the public with the above suggestions.

There should be a few good alternative anime models out there that are native trained and not inheriting from NovelAI running on RAIL license.

**Comment by Ihateyoudattebayo** [comment]
> > Regarding ethics: The final model release should have all of anything v3's weights replaced with standard stable diffusion. Using a variant of novelai is the most controversial part of the project due to its origin, so it would be best to release the model without its reliance. As you mentioned in the previous thread, I am glad to hear that this change will not affect the model greatly. The model should also use the same ethical licenses as Stable Diffusion does, e.g the RAIL license. Considering how beneficial this can be for artists, where it performs best under human-drawn lineart (and thus, human ideas), I see no problems releasing this to the public with the above suggestions.
> 
> There should be a few good alternative anime models out there that are native trained and not inheriting from NovelAI running on RAIL license.


The issue is that the best ones are inherited from novel ai. If only WD used SD 1.5 instead of 2.

**Comment by xITmasterx** [comment]
> > > Regarding ethics: The final model release should have all of anything v3's weights replaced with standard stable diffusion. Using a variant of novelai is the most controversial part of the project due to its origin, so it would be best to release the model without its reliance. As you mentioned in the previous thread, I am glad to hear that this change will not affect the model greatly. The model should also use the same ethical licenses as Stable Diffusion does, e.g the RAIL license. Considering how beneficial this can be for artists, where it performs best under human-drawn lineart (and thus, human ideas), I see no problems releasing this to the public with the above suggestions.
> > 
> > 
> > There should be a few good alternative anime models out there that are native trained and not inheriting from NovelAI running on RAIL license.
> 
> The issue is that the best ones are inherited from novel ai. If only WD used SD 1.5 instead of 2.

And that's where the issue lies. But there are new Anime models being released in droves, that finding a good Anime model that doesn't use Novel AI is actually going to be slightly hard due to the numbers. Like, there's a lot of them.

**Comment by Nyxeka** [comment]
IMO just release it the same way controlnet was - as a general-purpose "extension" to existing models. You don't even need to provide sd 1.4 or 1.5, or novelai or wd or anything like that, just provide raw model states of your relevant layers and allow for re-attaching it as a diff 🤷 (if it works the same way controlnet stuff does)

**Comment by Ihateyoudattebayo** [comment]
> IMO just release it the same way controlnet was - as a general-purpose "extension" to existing models. You don't even need to provide sd 1.4 or 1.5, or novelai or wd or anything like that, just provide raw model states of your relevant layers and allow for re-attaching it as a diff 🤷 (if it works the same way controlnet stuff does)

S2P should remain as a stand alone desktop program, as it is not related to stable diffusion.

**Comment by Nyxeka** [comment]
> > IMO just release it the same way controlnet was - as a general-purpose "extension" to existing models. You don't even need to provide sd 1.4 or 1.5, or novelai or wd or anything like that, just provide raw model states of your relevant layers and allow for re-attaching it as a diff 🤷 (if it works the same way controlnet stuff does)
> 
> S2P should remain as a stand alone desktop program, as it is not related to stable diffusion.

quoted from V5_Preview:

> The model is not trained from scratch. The initial weight comes from three sources: (1) some layers are initialized with borrowed weights from a wild mixture of many stable diffusion models, and we manually picked some best layers from many models [2], during many experiments. (2) some layers are initialized from scratch. (3) some layer weights are manually written by Lvmin, typically some special connection layers.

But fair enough. It would still be pretty awesome if this was released as-is with the model they put together. It seems extremely useful.

**Comment by NaughtDZ** [comment]
I know that previously your team had some doubts about nodal editing. Now with the advent of ComfyUI, a node-based editor based on stablediffusion, will this influence whether your team ends up developing a node-based editor or not?

**Comment by Ihateyoudattebayo** [comment]
> I know that previously your team had some doubts about nodal editing. Now with the advent of ComfyUI, a node-based editor based on stablediffusion, will this influence whether your team ends up developing a node-based editor or not?

A s2p which can add lighting and textures, needs a node system in order to achieve a studio level product.

Chainer or enchanr have a really good gui.

**Comment by blitzcaster** [comment]
The party is over! Nvidia, along with Shutterstock, Getty Images, and Adobe decided to release their own models. They claim to use ethical and legally sourced training dataset. This is a good reason for them to stomp down any open-source model and new start-up coming to AI party. The future of open-source is bleak on generative AI. Owner of massive data (read: massive corporations) will use artists emotion and empathy to stomp any emerging competition. Not a problem for them because they legally own their data.

**Comment by Ihateyoudattebayo** [comment]
> The party is over! Nvidia, along with Shutterstock, Getty Images, and Adobe decided to release their own models. They claim to use ethical and legally sourced training dataset. This is a good reason for them to stomp down any open-source model and new start-up coming to AI party. The future of open-source is bleak on generative AI. Owner of massive data (read: massive corporations) will use artists emotion and empathy to stomp any emerging competition. Not a problem for them because they legally own their data.

S2P is over 5 years old, and only recently started using Stable Diffusion as a base.

**Comment by blitzcaster** [comment]
> S2P is over 5 years old, and only recently started using Stable Diffusion as a base.

Look who is suing Stability AI now, Getty Images, they now play the same game. The difference is going forward, every open-source image generation project will easily get sued by claiming their data is tainted. Unlike corporations that control or own massive amounts of data, open-source project is at a disadvantage of having to purchase the data to improve their software. So basically, they gated the improvement of open-source project. They can set a high price to ensure others can't compete with them.

**Comment by Ihateyoudattebayo** [comment]
> > S2P is over 5 years old, and only recently started using Stable Diffusion as a base.
> 
> Look who is suing Stability AI now, Getty Images, they now play the same game. The difference is going forward, every open-source image generation project will easily get sued by claiming their data is tainted. Unlike corporations that control or own massive amounts of data, open-source project is at a disadvantage of having to purchase the data to improve their software. So basically, they gated the improvement of open-source project. They can set a high price to ensure others can't compete with them.

They would not be bothered in any way by an anime girl colouring system.

**Comment by xITmasterx** [comment]
> > > S2P is over 5 years old, and only recently started using Stable Diffusion as a base.
> > 
> > 
> > Look who is suing Stability AI now, Getty Images, they now play the same game. The difference is going forward, every open-source image generation project will easily get sued by claiming their data is tainted. Unlike corporations that control or own massive amounts of data, open-source project is at a disadvantage of having to purchase the data to improve their software. So basically, they gated the improvement of open-source project. They can set a high price to ensure others can't compete with them.
> 
> They would not be bothered in any way by an anime girl colouring system.

This is literally a open source coloring program/extension(?) whose sole purpose is to create colored works from line art. Doubt that whatever the corpos are doing is kind of similar to what Illyasviel's doing right now. 

Beside that point, was there any update from Illyasviel regarding this version of Style2Paints? It's been a month and a half since we last heard from her, and she's been busy with other extensions ever since then.

**Comment by Ihateyoudattebayo** [comment]
Ok guys, the line-art model was released on ControlNet v 1.1

Now we wait until the summer for an update
.

**Comment by NanoAi** [comment]
Hoping for an update on this soon.

**Comment by wpfnlp** [comment]
您的邮件已收到！

**Comment by Ihateyoudattebayo** [comment]
> Hoping for an update on this soon.

we wait

**Comment by My12123** [comment]
@lllyasviel 
Add the possibility of coloring based on the reference image (having a colored image of a character or scene, color it exactly the same, but on a similar image or the next image).

**Comment by qingxian233** [comment]
您好(*?0?7▽｀)ノノ青弦已经收到您的邮件啦！请耐心等待，10个工作日内会给您回复w

**Comment by Ihateyoudattebayo** [comment]
> @lllyasviel Add the possibility of coloring based on the reference image (having a colored image of a character or scene, color it exactly the same, but on a similar image or the next image).

I guess Lora could do that but on steroids, but it would be cool to have something similar.

**Comment by Ihateyoudattebayo** [comment]
bruh

**Comment by Ihateyoudattebayo** [comment]
we wait

**Comment by wpfnlp** [comment]
您的邮件已收到！

**Comment by Ihateyoudattebayo** [comment]
e

**Comment by wpfnlp** [comment]
您的邮件已收到！

**Comment by qingxian233** [comment]
您好(*?0?7▽｀)ノノ青弦已经收到您的邮件啦！请耐心等待，10个工作日内会给您回复w

**Comment by TruthSearcher** [comment]
@lllyasviel any update on stylepaint

**Comment by qingxian233** [comment]
您好(*?0?7▽｀)ノノ青弦已经收到您的邮件啦！请耐心等待，10个工作日内会给您回复w

**Comment by wpfnlp** [comment]
您的邮件已收到！

**Comment by Ihateyoudattebayo** [comment]
> Update:
> 
> The release of Style2Paints V5 will be delayed again. We need to make more thoughtful considerations about technology and the potential effects.
> 
> We are committed to presenting technology in a safe and beneficial way for all artists. We will let you know when we are ready. Thank you for your support and understanding!

It has been almost a year since this announcement, is s2p abandonned

**Comment by wpfnlp** [comment]
您的邮件已收到！

**Comment by krigeta** [comment]
> Update:
> 
> The release of Style2Paints V5 will be delayed again. We need to make more thoughtful considerations about technology and the potential effects.
> 
> We are committed to presenting technology in a safe and beneficial way for all artists. We will let you know when we are ready. Thank you for your support and understanding!

Please look into this, I know you are busy with stable diffusion but this is something else, please release the models

**Comment by Yerfina** [comment]
Hello, I would like to know if you managed to solve the ethical issues and you were ready to release the V5 models.

I am particularly interested in this project, because so far Dorothy is the only image AI I could find that really aims to preserve the original aspect of the lineart.
Don't get me wrong: I'm not an artist, and my linearts are mediocre at best, but I put a lot of thoughts into every detail. So far, I haven't found a single AI that was able to color and shade my lineart without completely destroying my original work.

So, to be honest, I care less for Alice, but I'm really interested in what Dorothy has to offer.

**Comment by wpfnlp** [comment]
您的邮件已收到！

**Comment by Ihateyoudattebayo** [comment]
> Hello, I would like to know if you managed to solve the ethical issues and you were ready to release the V5 models.
> 
> I am particularly interested in this project, because so far Dorothy is the only image AI I could find that really aims to preserve the original aspect of the lineart. Don't get me wrong: I'm not an artist, and my linearts are mediocre at best, but I put a lot of thoughts into every detail. So far, I haven't found a single AI that was able to color and shade my lineart without completely destroying my original work.
> 
> So, to be honest, I care less for Alice, but I'm really interested in what Dorothy has to offer.

They released the models as a form of control net. Though the old one from 4 years ago is still more effective

**Comment by Ihateyoudattebayo** [comment]
style2paints in the big 2025, but I think the other projects such as IC light are going to be used in the final style2paints v5 edition

**Comment by wpfnlp** [comment]
您的邮件已收到！

**Comment by scoolharis** [comment]
Any update regarding s2p?

**Comment by wpfnlp** [comment]
您的邮件已收到！

**Comment by jaimdd** [comment]
i know that V5 will NEVER be released but at least can you fix v4.5? it doesnt work at all i get an exception after selecting rtx 2080 as the strongest option, i have an rtx 3080

<img width="1332" height="592" alt="Image" src="https://github.com/user-attachments/assets/de913dae-e73c-445d-9bad-1f8ccea49322" />
